### DISCUSSION TITLE
Contribution do's and don'ts

### DISCUSSION THREAD

**Discussion by Gabriel-Ladzaretti** [renovatebot/renovate#15442] [Suggest an Idea]
I was thinking of creating some kind of a document to add to `renovatebot/renovate/tree/main/docs/development`.

This doc could contain best practices/do's & don'ts/tips and guide lines related to renovate development, this might make reviewers life easier and development faster.

From the top of my head, I would had for example the suggestion to use existing dependencies from our `package.json` instead of vanilla js/ts ones (i.e. use `got` for http requests, use `luxon` for date/time handling etc.).

WDYT?

**Comment by rarkins** [comment]
Yes, this would be great. I think "best practices" describes it best.

An important part of this would also be linking to examples where relevant. Sometimes people copy and paste or simply look at an existing module for inspiration and unfortunately find one which is *not* our current best practice, and then they get comments in a PR asking them to improve it (e.g. avoid unnecessary use of snapshots when an inline object match is better for understanding). So we should link to examples of good and bad if relevant.

Let's start small and iterate on it using reviews from PRs as the main input.
